src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -149,23 +159,29 @@ public void decorateCloneCommand(GitSCM scm, Run<?, ?> build, GitClient git, Tas
             // Git plugin does not support multiple independent repositories
             // in a single job definition.
             RemoteConfig rc = scm.getRepositories().get(0);
-            List<RefSpec> refspecs = rc.getFetchRefSpecs();
-            cmd.refspecs(refspecs);
+            cmd.refspecs(getRefSpecs(rc, env));
         }
         cmd.timeout(timeout);
 
-        Node node = GitUtils.workspaceToNode(git.getWorkTree());
-        EnvVars env = build.getEnvironment(listener);
-        Computer comp = node.toComputer();
-        if (comp != null) {
-            env.putAll(comp.getEnvironment());
-        }
         for (NodeProperty nodeProperty: node.getNodeProperties()) {
             nodeProperty.buildEnvVars(env, listener);
         }
         cmd.reference(env.expand(reference));
     }
 
+    @NonNull
+    private static String getParameterString(@CheckForNull String original, @NonNull EnvVars env) {
+        return env.expand(original);
+    }
```

Comment:
The `@NonNull` annotation does not seem to match the behavior of `env.expand()`.  I may be reading incorrectly, but it seems that `env.expand()` calls `Util.replaceMacro` and it seems to return null when the string argument is passed as a null.

Reply:
Copied that from GitSCM, will remove annotation but it could be removed from both places if it is incorrect.